metadata-app/src/components/ResourceUpload/validationSchema.js

```diff
@@ -10,7 +10,12 @@ export default yup.object().shape({
       .string()
       .url(
         'If you are linking to a dataset, please include "https://" at the beginning of your URL.'
-      ),
+      )
+      .when('id', (id, schema) => {
+        // if id exists meaning that it's in edit mode then pass the url validation,
+        // since in the form there upload field is hidden
```

Comment:
The comment is a little unclear, is this right?

> If the Id exists, then we are editing an existing resource and the URL field will not be complete in the form payload. Only validate that the URL is a string in this case.

Can you clarify why the URL field is a string, but not a valid URL in the edit case? Based on the comment, I'm expected the URL to be empty or missing.

Reply:
@adborden when we are in edit mode the url of the resource are pointed to `https://inventory.sandbox.datagov.us/dataset/{resourece_id}`, so the urls are still being validated. However when we are in dev mode we have url something similar to `http://localhost:5000/dataset/{DATASET_ID}` and in this case it doesn't pass. So it's better to pass the validation silently unless we are not giving an option to the user to edit it in edit mode.

Reply:
😕  Hmm, why doesn't `http://localhost:5000/dataset/{DATASET_ID}` pass validation? That looks like a valid URL to me?

Reply:
@adborden I guess there is no ".com" or ".somethingElse" in the url, that's why it doesn't pass. If you want I can create a custom validation using a regex, instead of realying `yup.url()` schema. When you confirm I can create a separate PR for this and will remove that code related `.when(...)`.

Reply:
@kmanaseryan @adborden here is the pattern used by `yup` package to validate URLs:

https://github.com/jquense/yup/blob/b53e5f23ced4c2df7636203e1300b9938fd33b3a/src/string.ts#L11

which fails for `http://localhost:5000`.

Reply:
In that case, we at least want to make a comment that we're changing the validation rule to only validate for a string in order to avoid validation failing for http://localhost in development. I would also create a bug for this that we can link to but icebox it for now. There should be a better way to handle it than changing the validation like this.

Reply:
I think we shouldn't disable validation and use `127.0.0.1` instead of `localhost` in development if we want to avoid validation failure. cc @kmanaseryan @adborden

Reply:
I'm fine with what @anuveyatsu suggests